crates/circuit/src/parameter/parameter_expression.rs

```diff
@@ -0,0 +1,2022 @@
+    pub fn tan(&self) -> Self {
+        Self {
+            expr: self.expr.tan(),
+            name_map: self.name_map.clone(),
+        }
+    }
+
+    /// Apply the arcsine to this expression; ``asin(self)``.
+    pub fn asin(&self) -> Self {
+        Self {
+            expr: self.expr.asin(),
+            name_map: self.name_map.clone(),
+        }
+    }
+
+    /// Apply the arccosine to this expression; ``acos(self)``.
+    pub fn acos(&self) -> Self {
+        Self {
+            expr: self.expr.acos(),
+            name_map: self.name_map.clone(),
+        }
+    }
+
+    /// Apply the arctangent to this expression; ``atan(self)``.
+    pub fn atan(&self) -> Self {
+        Self {
+            expr: self.expr.atan(),
+            name_map: self.name_map.clone(),
+        }
+    }
+
+    /// Exponentiate this expression; ``exp(self)``.
+    pub fn exp(&self) -> Self {
+        Self {
+            expr: self.expr.exp(),
+            name_map: self.name_map.clone(),
+        }
+    }
+
+    /// Take the (natural) logarithm of this expression; ``log(self)``.
+    pub fn log(&self) -> Self {
+        Self {
+            expr: self.expr.log(),
+            name_map: self.name_map.clone(),
+        }
+    }
+
+    /// Take the absolute value of this expression; ``|self|``.
+    pub fn abs(&self) -> Self {
+        Self {
+            expr: self.expr.abs(),
+            name_map: self.name_map.clone(),
+        }
+    }
+
+    /// Return the sign of this expression; ``sign(self)``.
+    pub fn sign(&self) -> Self {
+        Self {
+            expr: self.expr.sign(),
+            name_map: self.name_map.clone(),
+        }
+    }
+
+    /// Clone the expression.
+    pub fn copy(&self) -> Self {
+        // TODO: Is this necessary? Doesn't ParameterExpression.clone() implement the same thing?
+        Self {
+            expr: self.expr.clone(),
+            name_map: self.name_map.clone(),
+        }
+    }
+
+    /// Complex conjugate the expression.
+    pub fn conjugate(&self) -> Self {
+        Self {
+            expr: self.expr.conjugate(),
+            name_map: self.name_map.clone(),
+        }
+    }
+
+    /// Compute the derivative of the expression with respect to the provided symbol.
+    ///
+    /// Note that this keeps the name map unchanged. Meaning that computing the derivative
+    /// of ``x`` will yield ``1`` but the expression still owns the symbol ``x``. This is
+    /// done such that we can still bind the value ``x`` in an automated process.
+    pub fn derivative(&self, param: &Symbol) -> Result<Self, String> {
+        Ok(Self {
+            expr: self.expr.derivative(param)?,
+            name_map: self.name_map.clone(),
+        })
+    }
+
+    /// Expand the expression.
+    pub fn expand(&self) -> Self {
+        Self {
+            expr: self.expr.expand(),
+            name_map: self.name_map.clone(),
+        }
+    }
+
+    /// Substitute symbols with [PyParameterExpression]s.
```

Comment:
```suggestion
    /// Substitute symbols with [ParameterExpression]s.
```

Probably should do a replace-all for this.